src/Symfony/Component/Form/Extension/Core/Type/DateType.php

```diff
@@ -50,6 +50,10 @@ public function buildForm(FormBuilderInterface $builder, array $options)
             throw new InvalidOptionsException('The "format" option must be one of the IntlDateFormatter constants (FULL, LONG, MEDIUM, SHORT) or a string representing a custom format.');
         }
 
+        if (!empty($pattern) && !preg_match('/^([yMd]+)[^yMd]*([yMd]+)[^yMd]*([yMd]+)$/', $pattern)) {
```

Comment:
This forces the yMd letters to always be all there - but the check on L54-58 below allows only one of them.
Is this really legitimate?

Reply:
You are right, I've forgot something the Regexp has been updated.